homeassistant/components/miele/config_flow.py

```diff
@@ -71,3 +76,27 @@ async def async_oauth_create_entry(self, data: dict) -> ConfigFlowResult:
                 self._get_reconfigure_entry(), data=data
             )
         return await super().async_oauth_create_entry(data)
+
+    async def async_step_zeroconf(
+        self, discovery_info: ZeroconfServiceInfo
+    ) -> ConfigFlowResult:
+        """Prepare configuration for a zeroconf discovered Miele device."""
+        self.name = discovery_info.name.split(".", 1)[0]
+        # The unique_id here is needed for zeroconf discovery to behave properly
+        await self.async_set_unique_id(unique_id=UNIQUE_ID)
+        self._abort_if_unique_id_configured()
+
+        return await self.async_step_zeroconf_confirm()
+
+    async def async_step_zeroconf_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle a flow initiated by zeroconf."""
+
+        if user_input is not None:
+            return await self.async_step_user()
+
+        return self.async_show_form(
+            step_id="zeroconf_confirm",
+            description_placeholders={CONF_NAME: self.name},
+        )
```

Comment:
I think you can just use the default implementation of the zeroconf, as we don't have any use for the network information

Reply:
The only thing what you would be missing is that the discovery flow doesn't have the name of the device, but I think it also makes it more clear as someone might expect multiple discovery flows for their multiple devices

Reply:
I tested IRL and it works fine without a single line of code in config_flow.py. The only thing is you are directly sent to the Miele login screen without any confirmation. It is fine with me.